File: core/src/main/java/com/cc/core/actions/NullAction.java
package com.cc.core.actions;

public class NullAction implements Action {
    private String missingKey;
    public NullAction(){}
    public NullAction(String missingKey) {
        this.missingKey = missingKey;
    }
    @Override
    public ActionResult execute(Object... args) {
        return new ActionResult(false, "Could not find implementation for: '" + missingKey + "'");
    }

    @Override
    public String key() {
        return "NullAction";
    }

    public void setMissingKey(String missingKey) {
        this.missingKey = missingKey;
    }
}

File: core/src/main/java/com/cc/core/actions/NullAction.kt
package com.cc.core.actions

class NullAction : Action {
    private var missingKey: String? = null
    constructor()
    constructor(missingKey: String) {
        this.missingKey = missingKey
    }

    override fun execute(actionId : String, vararg args: Any?): ActionResult? {
        return ActionResult(actionId, false, "Could not find implementation for: '$missingKey'")
    }

    override fun key(): String? {
        return "NullAction"
    }

    fun setMissingKey(missingKey: String?) {
        this.missingKey = missingKey
    }
}